tuning/data/data_preprocessing_utils.py

```diff
@@ -96,3 +117,48 @@ def get_data_collator(
         raise ValueError(
             "Could not pick a data collator. Please refer to supported data formats"
         )
+
+class VisionDataCollator:
```

Comment:
could you move the collator to a separate file for keeping functionality separate?

Reply:
THanks fo the feedback @dushyantbehl all of the changes you suggest look good, I'm really interseted in the questions I listed above and tagged you on if you can take a look